src/aap_eda/core/validators.py

```diff
@@ -57,6 +59,49 @@ def check_if_de_exists(decision_environment_id: int) -> int:
 
 
 def check_if_de_valid(image_url: str, eda_credential_id: int):
+    parsed_url = urllib.parse.urlparse(image_url)
+    base_message = f"Image url {image_url} is malformed; "
+    if parsed_url.scheme:
```

Comment:
I'm concerned about the inclusion of the '@sha256:...' in this example. It doesn't appear that anything beyond a single occurrence matching [a-zA-Z0-9_][a-zA-Z0-9._-]{0,127} is valid as a tag (the OCI REST interface does allow for queries involving such). 

I'll have to check when I have better access, but if someone can point to a definitive statement on the topic before then that would be great.

Reply:
A tag with a digest is rejected when creating an image.  And while you can specify multiple tags the result is each tag is applied to a separate copy of the image.

Reply:
@jshimkus-rh The sha256 in the image url is a valid syntax. Its supported by podman-pull as well as podman-py

Reply:
OK.  Any others?  
My plan is then to split the tag off as being bounded by `:<tag>[@<everything-else>]`, validating the tag and passing everything else through.

Reply:
@jshimkus-rh Does the OCI spec talk about the tags in the image url format. We recently found an issue with podman-py where it can't handle port numbers in the image_url. If port numbers are not supposed to be defined in the image_url it might help us with the parsing.

Reply:
Docker describes what constitutes a valid specification here: https://docs.docker.com/reference/cli/docker/image/tag/. This includes a link to the OCI distribution spec whose section (beginning here https://github.com/opencontainers/distribution-spec/blob/8376368dd8aadc33bf6c88a8b765df90287bb5c8/spec.md?plain=1#L147) talks about the same thing though less straightforwardly. 
The OCI spec provides regexes for both the <path/to/image> and <tag> which comport with Docker's specification. It also, rather tangentially, talks about <host>[:port] (another case where Docker is clearer on the topic). Regardless, both agree that the composite specification is <host>[:port]/<path/to/image>[:<tag>]. If podman code isn't handling that then it's not standard compliant.